From 0582c76ce9c35ce8d49cba598e0e17073dd875b5 Mon Sep 17 00:00:00 2001
From: Alexandru Gagniuc <mr.nuke.me@gmail.com>
Date: Fri, 23 Sep 2022 08:30:03 -0500
Subject: [PATCH 09/14] Revert "qca-ssdk: remove bridge fdb entry for the
 authentication failed mac"

This change causes an undefined reference to "br_fdb_delete_by_netdev".
This reverts commit 144f02b982c8c707aaf84b57d8c277d03d877236.

Signed-off-by: Alexandru Gagniuc <mr.nuke.me@gmail.com>
---
 src/ref/ref_acl.c | 11 -----------
 1 file changed, 11 deletions(-)

--- a/src/ref/ref_acl.c
+++ b/src/ref/ref_acl.c
@@ -23,7 +23,6 @@
 #include "ssdk_init.h"
 #include "ssdk_plat.h"
 #include <linux/etherdevice.h>
-#include <linux/if_bridge.h>
 
 /* entry 0-1 is for global deny all and accept eapol rule
    entry 2-9 is for phy port1 specific mac accept rule
@@ -128,7 +127,6 @@ _ref_acl_mac_entry_create_rule(a_uint32_
 {
 	sw_error_t rv = SW_OK;
 	fal_acl_rule_t rule = {0};
-	struct net_device *eth_dev = NULL;
 	a_uint32_t port_id = ssdk_ifname_to_port(dev_id, entry->ifname);
 	SSDK_DEBUG("port_id %d entry_idx %d\n", port_id, entry_idx);
 
@@ -224,15 +222,6 @@ _ref_acl_mac_entry_create_rule(a_uint32_
 		ref_acl_mac_entry[dev_id][entry_idx].port_map = BIT(port_id);
 		ref_acl_mac_entry[dev_id][entry_idx].acl_policy = 1;
 	}
-	else if (!is_deny_all_mac(entry->src_mac.uc) && !entry->acl_policy)
-	{
-		eth_dev = dev_get_by_name(&init_net, entry->ifname);
-		if (eth_dev)
-		{
-			br_fdb_delete_by_netdev(eth_dev, entry->src_mac.uc, 0);
-			dev_put(eth_dev);
-		}
-	}
 	return rv;
 }
 
